### PR TITLE
[MOOSE-8] Styleguide inline links buttons

### PR DESCRIPTION
**Pull request by jamiepaul**
## What does this do/fix?

- Adds `postcss-inline-svg` for icons
- Adds default button element styles in theme.json
- Adds custom styles to core Button block (primary, secondary, ghost)
- Adds styles for inline links

## QA

Links to relevant issues
- [Link to MOOSE-8](https://moderntribe.atlassian.net/browse/MOOSE-8)

Screenshots:

### CODE REVIEW

File: wp-content/plugins/core/src/Blocks/Blocks_Definer.php [hunk 1]
```diff
@@ -20,6 +22,7 @@ public function define(): array {
 			] ),
 
 			self::STYLES          => DI\add( [
+				// DI\get( Button::class ),
```
Comment (jamiepaul) [inline wp-content/plugins/core/src/Blocks/Blocks_Definer.php:25]: @dpellenwood I'm sure I'm doing something wrong w/ namespaces but I couldn't seem to get my custom styles for the Button block to register. Can I get a hand w/ this when you have a min?

File: wp-content/themes/core/assets/pcss/typography/anchors.pcss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+}
+
+/* -------------------------------------------------------------------------
+ * Anchor: Inline Links
+ *
+ * Not in theme.json as the selector needs to be more specific
+ * ------------------------------------------------------------------------- */
+
+:where(p, li) a {
```
Comment (jamiepaul) [inline wp-content/themes/core/assets/pcss/typography/anchors.pcss:25]: `theme.json` targets very generically when setting default element styles. In addition, the last few projects needed too many :not() selectors to override the fancy inline link styles, I think it's best to target this way.

See [example](https://github.com/wpcomvip/qchi-lendnation/blob/9fce7ba195bd29e5779adf8311c11c471f152213/themes/core/assets/pcss/global/links.pcss#L1) from LendNation

File: wp-content/themes/core/parts/header.html [hunk 3]
```diff
@@ -1 +1,10 @@
-Header
+<!-- wp:group {"layout":{"type":"constrained"}} -->
+<div class="wp-block-group">
+	<!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"bottom":"var:preset|spacing|40"}}},"layout":{"type":"flex","justifyContent":"space-between"}} -->
+	<div class="wp-block-group alignwide" style="padding-bottom:var(--wp--preset--spacing--40)">
+		<!-- wp:site-title {"level":0} /-->
+		<!-- wp:navigation {"layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right"}} /-->
```
Comment (jamiepaul) [inline wp-content/themes/core/parts/header.html:6]: Added this purely to see how my button and link styles affected the site title and nav blocks. I can remove it, if needed.

File: wp-content/plugins/core/src/Blocks/Blocks_Definer.php (hunk 1 above)
Comment (dpellenwood) [reply wp-content/plugins/core/src/Blocks/Blocks_Definer.php]: @jamiepaul , this should be fixed now. You may have to run `composer dump-autoload` after pulling the latest into your local branch.

File: postcss.config.js [hunk 4]
```diff
@@ -46,6 +47,7 @@ const plugins = [
 		{
 			files: [
 				`${ pkg.config.coreThemeDir }/assets/pcss/custom-selectors/_variables.pcss`,
+				`${ pkg.config.coreThemeDir }/assets/pcss/icons/_variables.pcss`,
```
Comment (jamiepaul) [inline postcss.config.js:50]: Needed for `svg-load()` and `svg-inline()` functions to work in other entry points

File: wp-content/themes/core/blocks/core/button/style.pcss [hunk 5]
```diff
@@ -1,6 +1,104 @@
...
+
+.wp-block-button__link {
+	transition: var(--transition);
+}
+
+/* -------------------------------------------------------------------------
+ * Button: Arrow After Label
+ * ------------------------------------------------------------------------- */
+
+:is(.is-style-primary, .is-style-secondary) .wp-block-button__link {
```
Comment (dpellenwood) [inline wp-content/themes/core/blocks/core/button/style.pcss:13]: Any chance we can set up the the styles as parent-selectors (nested) throughout this file? Something like:

```
.wp-block-button__link {
	
		:is(.is-style-primary, .is-style-secondary) & { 
			...
		}
```

File: wp-content/themes/core/assets/pcss/typography/anchors.pcss [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * Anchor: Defaults
+ * ------------------------------------------------------------------------- */
+
+/* -------------------------------------------------------------------------
+ * Anchor: Inline Links
+ *
+ * Not in theme.json as the selector needs to be more specific
+ * ------------------------------------------------------------------------- */
+
+:where(p, li) a {
```
Comment (dpellenwood) [inline wp-content/themes/core/assets/pcss/typography/anchors.pcss:17]: Similar here re: nesting the conditional selectors inside the anchor?

File: wp-content/themes/core/blocks/core/button/style.pcss (hunk 5 above)
Comment (jamiepaul) [reply wp-content/themes/core/blocks/core/button/style.pcss:13]: Yes, will do 👍

File: wp-content/themes/core/theme.json [hunk 7]
```diff
@@ -202,6 +202,26 @@
...
+				},
+				"typography": {
+					"fontFamilies": [],
+					"fontSizes": [],
+					"textTransform": false
+				}
+			},
+			"core/group": {
+				"color": {
+					"background": true
```
Comment (jamiepaul) [inline wp-content/themes/core/theme.json:222]: Added this so we can test elements/blocks on a background easier, like [this](http://p.tri.be/i/kbORAK)

File: wp-content/themes/core/assets/pcss/typography/anchors.pcss [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/* -------------------------------------------------------------------------
+ *
+ * Typography: Anchors
+ *
+ * ------------------------------------------------------------------------- */
+
+a {
+
+	&:focus-visible {
```
Comment (jamiepaul) [inline wp-content/themes/core/assets/pcss/typography/anchors.pcss:9]: Still exploring a couple other ways of implementing focus-vislble in a way that meets contrast requirements on any background/element color. Doing that alongside the reset in my next PR

File: .stylelintrc.json [hunk 9]
```diff
@@ -4,9 +4,10 @@
 		"at-rule-no-unknown": [
 			true,
 			{
-				"ignoreAtRules": [ "define-mixin", "mixin" ]
+				"ignoreAtRules": [ "define-mixin", "mixin", "svg-load" ]
 			}
 		],
-		"selector-class-pattern": null
+		"selector-class-pattern": null,
```
Comment (dpellenwood) [inline .stylelintrc.json:10]: Curious what this rule ignore solved?

**Review by dpellenwood — APPROVED** [review]
This looks great, Jamie. I assume the TODO's in the PR description are no longer relevant?

File: .stylelintrc.json (hunk 9 above)
Comment (jamiepaul) [reply .stylelintrc.json:10]: That one was pre-existing, the line changed came from the comma needed for the next rule

**Pull request merged**